src/site/content/en/blog/serial/index.md

```diff
@@ -0,0 +1,610 @@
+the widget, so that by the time it gets to its final destination, it's a fully
+functioning widget.
+
+<figure class="w-figure">
+  <img src="./aeroplane-factory.jpg" alt="Photo of an aeroplane factory">
+  <figcaption class="w-figcaption">World War 2 Castle Bromwich Aeroplane Factory</figcaption>
+</figure>
+
+For example, consider how to create a transform stream class that consumes a
+stream and chunks it based on line breaks. Its `transform()` method is called
+every time new data is received by the stream. It can either enqueue the data or
+save it for later. The `flush()` method is called when the stream is closed, and
+it handles any data that hasn't been processed yet.
+
+To use the transform stream class, you need to pipe an incoming stream through
+it. In the readableStream example under [Read from a serial port](#read-port),
+the original input stream was only piped through a `TextDecoderStream`, so we
+need to call `pipeThrough()` to pipe it through our new `LineBreakTransformer`.
+
+```js
+class LineBreakTransformer {
+  constructor() {
+    // A container for holding stream data until a new line.
+    this.chunks = "";
+  }
+
+  transform(chunk, controller) {
+    // Append new chunks to existing chunks.
+    this.chunks += chunk;
+    // For each line breaks in chunks, send the parsed lines out.
+    const lines = this.chunks.split("\r\n");
+    this.chunks = lines.pop();
+    lines.forEach((line) => controller.enqueue(line));
+  }
+
+  flush(controller) {
+    // When the stream is closed, flush any remaining chunks out.
+    controller.enqueue(this.chunks);
+  }
+}
+```
+
+```js/2-5
+const textDecoder = new TextDecoderStream();
+const readableStreamClosed = port.readable.pipeTo(textDecoder.writable);
+const reader = textDecoder.readable
+  .pipeThrough(new TransformStream(new LineBreakTransformer()))
+  .getReader();
+```
+
+For debugging serial device communication issues, use the `tee()` method of
+`port.readable` to split the streams going to or from the serial device. The two
+streams created can be consumed independently and this allows you to print one
+to the console for inspection.
+
+```js
+const [appReadable, devReadable] = port.readable.tee();
+
+// TODO: Update UI with incoming data from appReadable.
+// TODO: Log incoming data in JS console for inspection from devReadable.
+```
+
+## Codelab {: #codelab }
+
+In the [Google Developer codelab], you'll use the Serial API to interact with a
+[BBC micro:bit] board to show images on its 5x5 LED matrix.
+
+## Polyfill {: #polyfill }
+
+On Android, support for USB-based serial ports is possible using the WebUSB API
+and the [Serial API polyfill]. This polyfill is limited to hardware and
+platforms where the device is accessible via the WebUSB API because it has not
+been claimed by a built-in device driver.
+
+## Security and privacy {: #security-privacy }
+
+The Chrome team has designed and implemented the Serial API using the core
```

Comment:
```suggestion
The spec authors designed and implemented the Serial API using the core
```

Reply:
Fixed. But what is the rationale?

Reply:
It seems that Reilly listed spec authors in a way that credits everyone who contributed to the spec repo in some way. Not all of them are Google employees.